### PR TITLE
Add a development guide for building provider images locally

### PR DESCRIPTION
**Pull request by turkenf**
### Description of your changes

This PR adds a guide for building provider images locally.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

Manually tested in this PR: https://github.com/upbound/provider-gcp/pull/361

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: docs/building-provider-images.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+    ```yaml
+    apiVersion: pkg.crossplane.io/v1
+    kind: Provider
+    metadata:
+        name: upbound-provider-gcp-config
+    spec:
+        package: turkenf/provider-gcp-config-amd64:v0.36.0-rc.0.9.g6940237d
+        skipDependencyResolution: true
+    ---
```
Comment (ulucinar) [inline docs/building-provider-images.md:74]: If we implement the suggestion [here](https://github.com/upbound/provider-gcp/pull/361/files#r1297066558), we may also get rid of the explicit dependency installation here. 

I've some concerns surfacing the dependency installation here.

Comment (turkenf) [reply docs/building-provider-images.md:74]: Please see: https://github.com/upbound/provider-gcp/pull/361#discussion_r1297652261

File: docs/building-provider-images.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
+# Building Provider Images Locally for Family Providers
```
[the remark this answers is not in the document]
Comment (mbbush) [reply docs/building-provider-images.md:1]: > Building the Official Provider Family Images Locally
sounds like more natural English to me (from the USA)

**Comment by jeanduplessis** [comment]
@turkenf I agree with Alper that we should move this information to https://docs.upbound.io
I will close this PR.

**Pull request closed without merging**